gk/main.c

```diff
@@ -417,21 +425,141 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 		goto flow_hash;
 	}
 
+	ret = snprintf(ring_name, sizeof(ring_name), "gk_ring_%d", block_idx);
+	RTE_ASSERT(ret < sizeof(ring_name));
+
+	instance->cmd_ring = (struct rte_ring *)rte_ring_create(ring_name,
+		gk_conf->flow_ht_size / 2, socket_id, RING_F_SC_DEQ);
```

Comment:
I think that having all mailboxes defaulting to 128 entries is fine for now. Make it a define, and remember to add a comment explaining the meaning and limitations, for example, `rte_ring_create()` requires is to be a power of 2 according to the link below:

http://dpdk.org/doc/api-16.04/rte__ring_8h.html#a5556487d8ec76cacfc059b66d8b77c98

Moreover, the link above also mentions that the maximum size is count - 1. All this information should go into the comment of the new define.

Suggestion for the define name: MAILBOX_MAX_ENTRIES.